### PR TITLE
[stable/datadog] add trace agent mounts

### LINKED ISSUE #21586: [stable/datadog] v2 of chart no longer supports unix socket trace agent
I use unix sockets for my APM to keep ports private. This is supported by the agent with `DD_APM_RECEIVER_SOCKET` set to something like `"/var/run/datadog/apm.socket"`

The v2 of the chart no longer supports volume mounts for the APM agent.  Please modify [container-trace-agent.yaml](https://github.com/helm/charts/blob/master/stable/datadog/templates/container-trace-agent.yaml) to allow volume mounts, similar to how container agent has
```
{{- if .Values.agents.volumeMounts }}
{{ toYaml .Values.agents.volumeMounts | indent 4 }}
{{- end }}
```

### PR DESCRIPTION
**Pull request by cep21**
Resolves #21586

### CODE REVIEW

**Comment by cep21** [comment]
CC: @ahmed-mez

**Comment by ahmed-mez** [comment]
Thanks @cep21 ! cc @clamoriniere

File: stable/datadog/templates/container-trace-agent.yaml [hunk 1]
```diff
@@ -37,6 +37,10 @@
       mountPath: {{ template "datadog.confPath" . }}/datadog.yaml
       subPath: datadog.yaml
     {{- end }}
+    {{- if .Values.datadog.dogstatsd.useSocketVolume }}
+    - name: dsdsocket
+      mountPath: "/var/run/datadog"
+    {{- end }}
```
Comment (clamoriniere) [inline stable/datadog/templates/container-trace-agent.yaml:43]: This is the socket used by the dogstatsd server. 

I think it makes sens to add the new parameters `.Values.datadog.apm.useSocketVolume|sockerPath` 
```
datadog:
  apm:
    ## @param useSocketVolume - boolean - optional
    ## Enable APM over Unix Domain Socket
    ## ref: https://docs.datadoghq.com/agent/kubernetes/apm
    #
    # useSocketVolume: true

    ## @param socketPath - string - optional
    ## Path to the Trace-Agent socket
    #
    # socketPath: /var/run/datadog/apm.socket

The EnvVar `DD_APM_RECEIVER_SOCKET ` can also be added to the `trace-agent` container when `.Values.datadog.apm.useSocketVolume` is `true`.

WDYT?

**Review by clamoriniere — CHANGES_REQUESTED** [review]
Hi @cep21 
Thank you for opening this PR.
Using the trace-agent with a Unix-socket is not something that we advertise often, but still, it is something that we want to support. So, it wasn't on purpose that we remove this possibility.

I added a comment about how the fix can be done to allow this activation as we do for the Dogstatsd server. Let me know if you have any remark.

Also please update the `Chart.yaml` file (bump the version) and please describe your change in the `CHANGELOG.md` file.

File: stable/datadog/templates/container-trace-agent.yaml (hunk 1 above)
Comment (cep21) [reply stable/datadog/templates/container-trace-agent.yaml:43]: That sounds good to me.  I noticed however that the helm chart hard codes `/var/run/datadog` as the mount.  If someone used `/var/datadog/dsd` as the path instead, I don't think the mount part would work right.  Maybe also document this limitation?

Comment (clamoriniere) [reply stable/datadog/templates/container-trace-agent.yaml:43]: yes. it requires to have a new parameter for allowing the overwrite the hostpath. If you feel confortable you can also add this in your PR.

Also I think it makes sense to add a new "Volume.hostPath" for the trace-agent like it is done here for dogstatsd:
https://github.com/helm/charts/blob/master/stable/datadog/templates/daemonset-volumes-linux.yaml#L8

**Comment by clamoriniere** [comment]
Hi @cep21 
Friendly ping to know if you plan to rework this PR in the coming days?
If you prefer, I can take care of doing it.

**Comment by cep21** [comment]
@clamoriniere Please feel free to take over :)  I'm a bit busy and my current fork is working well, but I would love to switch back to the official helm chart.

**Pull request closed without merging**